.github/workflows/jacoco_check.yml

```diff
@@ -0,0 +1,88 @@
+#
+# Copyright 2021 ABSA Group Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+name: JaCoCo report
+
+on:
+  pull_request:
+    branches: [ master ]
+    types: [ opened, edited, synchronize, reopened ]
+
+jobs:
+  test:
+    name: Build and test
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        include:
+          # The project supports Scala 2.11, 2.12 and the default version of spark2
+          #          # The CI runs all tests suites for all supported Scala versions at build.yml
+          #          # The codebase for all Scala versions is the same, so the coverage is calculated only once
+          #          # Scala 2.12 is chosen since it is supported by the most wide range of Spark versions and
+          #          # vendor distributions.
+          - scala: 2.12.12
+            scalaShort: "2.12"
+            overall: 80.0
+            changed: 80.0
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Setup Scala
+        uses: olafurpg/setup-scala@v10
+        with:
+          java-version: "adopt@1.8"
+      - name: Build and run tests
+        run: sbt ++${{matrix.scala}} jacoco
+     #  Agent and server module code coverage
```

Comment:
This one is only for agent, not server. That is below as you wrote, so this comment here is a bit misleading